provider/pkg/await/await.go

```diff
@@ -491,9 +499,69 @@ func ssaUpdate(c *UpdateConfig, liveOldObj *unstructured.Unstructured, client dy
 	return currentOutputs, nil
 }
 
+// handleSSAIgnoreFields handles updating the inputs to either drop fields that are present on the cluster and not managed
+// by the current field manager, or to set the value of the field to the last known value applied to the cluster.
+func handleSSAIgnoreFields(c *UpdateConfig, liveOldObj *unstructured.Unstructured) error {
+	managedFields := liveOldObj.GetManagedFields()
+	managedFieldsSet := fieldpath.NewSet()
+
+	for _, f := range managedFields {
+		if f.Manager == c.FieldManager {
+			// We don't need to keep track of fields that we are currently managing.
+			continue
+		}
+		s := fieldpath.NewSet()
+		if err := s.FromJSON(bytes.NewReader(f.FieldsV1.Raw)); err != nil {
+			return fmt.Errorf("unable to parse managed fields from resource %q: %w", c.Inputs.GetName(), err)
+		}
```

Comment:
Nit: consider using `fluxssa.FieldsToSet(f.FieldsV1)`.

Reply:
Normally I prefer to reimplement upstream libraries to do simple tasks like this, but since we already import and use `fluxssa` I've updated the code to use this function instead.